### PR TITLE
allow page injection from outside fluentlenium

### PR DESCRIPTION
**Pull request by ataastar**
FluentAdapter.initTest run properly if page was injected already

### CODE REVIEW

File: fluentlenium-core/src/test/java/org/fluentlenium/integration/InjectedPageTest.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+import org.fluentlenium.core.annotation.Page;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.htmlunit.HtmlUnitDriver;
+
+public class InjectedPageTest extends FluentAdapter {
+
+    private final WebDriver driver = new HtmlUnitDriver(true);
```
Comment (filipcynarski) [inline fluentlenium-core/src/test/java/org/fluentlenium/integration/InjectedPageTest.java]: Why HTMLUnit driver is hardcoded here. I'd like to user every driver which is available for Selenium.

File: fluentlenium-assertj/src/test/java/org/fluentlenium/integration/localtest/LocalFluentCase.java [hunk 2]
```diff
@@ -29,7 +29,7 @@ public WebDriver getDefaultDriver() {
     }
 
     public static String getPath() {
-        String currentDir = System.getProperty("user.dir");
+        String currentDir = System.getProperty("user.dir").replace('\\', '/');
```
Comment (filipcynarski) [inline fluentlenium-assertj/src/test/java/org/fluentlenium/integration/localtest/LocalFluentCase.java]: I do not understand why are you doing such replacement? Java has mechanism to recognize what OS is running on and what is the path separator.

File: fluentlenium-core/src/main/java/org/fluentlenium/core/FluentAdapter.java [hunk 3]
```diff
@@ -98,28 +107,38 @@ private void injectPageIntoContainer(Fluent container)
         return container;
     }
 
-
     protected <T extends FluentPage> T initClass(Class<T> cls, Object... params) {
+        T page = null;
```
Comment (filipcynarski) [inline fluentlenium-core/src/main/java/org/fluentlenium/core/FluentAdapter.java]: What happened if the page is not set? Are you returning null object and NPE will be thrown on related methods?

File: fluentlenium-core/src/test/java/org/fluentlenium/integration/localtest/LocalFluentCase.java [hunk 4]
```diff
@@ -29,7 +29,7 @@ public WebDriver getDefaultDriver() {
         return new HtmlUnitDriver(true);
     }
     public static String getPath() {
-        String currentDir = System.getProperty("user.dir");
+        String currentDir = System.getProperty("user.dir").replace('\\', '/');
```
Comment (filipcynarski) [inline fluentlenium-core/src/test/java/org/fluentlenium/integration/localtest/LocalFluentCase.java]: As in previous classes I don't like such replacement. What is a reason to having that in the code?

**Comment by Toilal** [comment]
Merged with https://github.com/FluentLenium/FluentLenium/commit/4013ec54fdf8c07c9e745140ff7fea523b621cef

**Pull request closed without merging**